src/components/prometheusmetrics.ts

```diff
@@ -173,13 +177,25 @@ export class PrometheusMetrics {
         });
     }
 
+    /**
+     * Fetch metrics from all configured collectors
+     */
+    public async refresh () {
```

Comment:
Does `refresh` need a lock (or wrapper returning an ongoing promise) so it cannot get run more than once at a time?

Reply:
I think we're okay in this case. The caller should manage that, whether that be a bridge or the application requesting the /metrics endpoint. I don't think a race will cause any serious problems as you're just asking it to refresh metrics in collections, which should be RO functions.